### ISSUE TITLE
Applications & Processes Tabs are Blank with No Items

### ISSUE THREAD

**Issue by cameronbosch** [nokyan/resources#43]
Hi,, I'm running the flatpak version of Resources, and this is what I get in the Applications & Processes tabs:

![image](https://github.com/nokyan/resources/assets/72839795/34c7bd62-2f47-405c-b90e-d32b97197be5)

![image](https://github.com/nokyan/resources/assets/72839795/66b09f55-e2b1-4f8e-8c89-d14ed5ce352b)

Is there something in Flatseal I need to change or is this a legitimate bug?

Terminal output:
```
 INFO  resources::application > Resources (net.nokyan.Resources)
 INFO  resources::application > Version: 1.1.0 ()
 INFO  resources::application > Datadir: /app/share/resources
 DEBUG resources::application::imp > GtkApplication<Application>::startup

(resources:2): Gtk-WARNING **: 19:01:00.612: Unknown key gtk-modules in /home/cameron/.var/app/net.nokyan.Resources/config/gtk-4.0/settings.ini
 DEBUG resources::application::imp > GtkApplication<Application>::activate
thread 'main' panicked at src/ui/pages/cpu.rs:275:18:
already borrowed: BorrowMutError
stack backtrace:
   0:     0x55a5272f8d2c - <unknown>
   1:     0x55a52732705c - <unknown>
   2:     0x55a5272f4efe - <unknown>
   3:     0x55a5272f8b14 - <unknown>
   4:     0x55a5272fa5a3 - <unknown>
   5:     0x55a5272fa2bc - <unknown>
   6:     0x55a5272fab99 - <unknown>
   7:     0x55a5272faa97 - <unknown>
   8:     0x55a5272f9256 - <unknown>
   9:     0x55a5272fa7e2 - <unknown>
  10:     0x55a527324d13 - <unknown>
  11:     0x55a52732520a - <unknown>
  12:     0x55a5270ff19d - <unknown>
  13:     0x55a526fff55a - <unknown>
  14:     0x55a5270a4a13 - <unknown>
  15:     0x55a5272c656d - <unknown>
  16:     0x55a5272c3d9b - <unknown>
  17:     0x7fac3971ad07 - <unknown>
  18:     0x7fac3971ce17 - <unknown>
  19:     0x7fac3971d4d3 - g_main_context_iteration
  20:     0x7fac39900a4d - g_application_run
  21:     0x55a52700f496 - <unknown>
  22:     0x55a52700f596 - <unknown>
  23:     0x55a52712a69c - <unknown>
  24:     0x55a52708032b - <unknown>
  25:     0x55a526ffbc53 - <unknown>
  26:     0x55a526ffbc29 - <unknown>
  27:     0x55a5272ee24b - <unknown>
  28:     0x55a526ffbc9c - <unknown>
  29:     0x7fac3895208a - <unknown>
  30:     0x7fac3895214b - __libc_start_main
  31:     0x55a526ffbb55 - <unknown>
  32:                0x0 - <unknown>
```

The system is an Arch Linux system and Resources was installed via Flathub.

**Comment by sokorototo** [comment]
You are on KDE right? You could maybe try installing directly from the AUR? This could be a Flatpak issue

**Comment by nokyan** [comment]
Hi,
this is a bug though I really don't know how that occurs. Does it happen every time you start Resources? What CPU do you use?

**Comment by cameronbosch** [comment]
Hi, I use an AMD Ryzen 7950X CPU (16 cores, 32 threads).

**Comment by cameronbosch** [comment]
Hello. I am wondering if you have found the solution to this problem. If so, have you started work on a fix? If not, are you still looking for the problem?

I don't mean to be annoying, but I haven't seen a response for about a week, and I am just checking in.

**Comment by nokyan** [comment]
Hi, sorry for not responding to you, I've been quite busy the last few days. :)
So far I haven't been able to reproduce the issue, at least not in a KDE neon VM. I think I'll be able to at least not make it partially crash due to BorrowMutError though.
Have you tried building Resources natively (or using the AUR package, if that works) and testing that?

**Comment by freaktechnik** [comment]
Hi, I'm having the same issue, but on Fedora with Gnome. However I do have a 7950X as well, so I think that might be the issue here.

```
[martin@martin-desktop comm]$ flatpak run net.nokyan.Resources 
 INFO  resources::application > Resources (net.nokyan.Resources)
 INFO  resources::application > Version: 1.1.0 ()
 INFO  resources::application > Datadir: /app/share/resources
 DEBUG resources::application::imp > GtkApplication<Application>::startup
 DEBUG resources::application::imp > GtkApplication<Application>::activate
thread 'main' panicked at src/ui/pages/cpu.rs:275:18:
already borrowed: BorrowMutError
stack backtrace:
   0:     0x55ce5fd4fd2c - <unknown>
   1:     0x55ce5fd7e05c - <unknown>
   2:     0x55ce5fd4befe - <unknown>
   3:     0x55ce5fd4fb14 - <unknown>
   4:     0x55ce5fd515a3 - <unknown>
   5:     0x55ce5fd512bc - <unknown>
   6:     0x55ce5fd51b99 - <unknown>
   7:     0x55ce5fd51a97 - <unknown>
   8:     0x55ce5fd50256 - <unknown>
   9:     0x55ce5fd517e2 - <unknown>
  10:     0x55ce5fd7bd13 - <unknown>
  11:     0x55ce5fd7c20a - <unknown>
  12:     0x55ce5fb5619d - <unknown>
  13:     0x55ce5fa5655a - <unknown>
  14:     0x55ce5fafba13 - <unknown>
  15:     0x55ce5fd1d56d - <unknown>
  16:     0x55ce5fd1ad9b - <unknown>
  17:     0x7fa761dd5d07 - <unknown>
  18:     0x7fa761dd7e17 - <unknown>
  19:     0x7fa761dd84d3 - g_main_context_iteration
  20:     0x7fa761fbba4d - g_application_run
  21:     0x55ce5fa66496 - <unknown>
  22:     0x55ce5fa66596 - <unknown>
  23:     0x55ce5fb8169c - <unknown>
  24:     0x55ce5fad732b - <unknown>
  25:     0x55ce5fa52c53 - <unknown>
  26:     0x55ce5fa52c29 - <unknown>
  27:     0x55ce5fd4524b - <unknown>
  28:     0x55ce5fa52c9c - <unknown>
  29:     0x7fa76103b08a - <unknown>
  30:     0x7fa76103b14b - __libc_start_main
  31:     0x55ce5fa52b55 - <unknown>
  32:                0x0 - <unknown>
```

I'll see if I can figure out building it locally.

**Comment by cameronbosch** [comment]
Yeah, the high amount of CPU threads might be the culprit here. (The VirtualBox VM I used only had 1 or 2 threads so that may have been why it worked there but not with all 32 threads active.)

**Comment by freaktechnik** [comment]
Local flatpak build still crashes, probably with the same error just that it's now https://github.com/nokyan/resources/blob/bde509dccedf01b29d06622fa7c5c0cd96c9a284/src/ui/pages/cpu.rs#L292

**Comment by freaktechnik** [comment]
This simple change:
```iff --git a/src/ui/pages/cpu.rs b/src/ui/pages/cpu.rs
index 72c16e1..2c288de 100644
--- a/src/ui/pages/cpu.rs
+++ b/src/ui/pages/cpu.rs
@@ -191,7 +191,6 @@ impl ResCPU {
             let imp = this.imp();
 
             let logical_cpus = cpu_info.logical_cpus.unwrap_or(0);
-            imp.logical_cpus_amount.set(logical_cpus);
 
             imp.old_total_usage.set(cpu::get_cpu_usage(None).await.unwrap_or((0, 0)));
             *imp.old_thread_usages.borrow_mut() = Vec::with_capacity(logical_cpus);
@@ -199,6 +198,7 @@ impl ResCPU {
             for i in 0..logical_cpus {
                 imp.old_thread_usages.borrow_mut().push(cpu::get_cpu_usage(Some(i)).await.unwrap_or((0, 0)));
             }
+            imp.logical_cpus_amount.set(logical_cpus);
 
             imp.total_cpu.set_title_label(&i18n("CPU"));
             imp.total_cpu.set_subtitle(&i18n("N/A"));
```
avoids the crash, suggesting the widgets setup and first page refresh are racing.

**Comment by nokyan** [comment]
Thank you so much for your help. I'll include this fix along with some failsafes for the next release. :)

**Comment by nokyan** [comment]
Closing this as it's implemented in Resources 1.2. If there's still something wrong, don't hesitate to reopen. :)
